docs/architecture.md

```diff
@@ -1,8 +1,94 @@
 # Project Helix Architecture
 
-![](./Helix%20Architecture/Helix%20Stack.png)
 
-## Preface
+
+## Architectural Evolution
+
+Helix has undergone some architectural evolution since the start of the project. The main phases are:
+
+1. Helix 1 – a bring-your-own (Adobe I/O) runtime framework-driven approach that would allow server-side customization
+2. [Helix 2](#helix-2) – a software as a service that relies on client-side customization
+3. [Helix 3](#helix-3) – a software as a service with superior availability and performance with the same feature set as Helix 2
+
+The current version of the architecture is Helix 3, prior versions stay documented as long as they are in active use.
+
+## Helix 3
+
+Helix 3 delivers the core feature set of Helix (high-performance websites with low code and zero-friction authoring) in a simplified architecture that optimizes stability and availability.
+
+![](./Helix%20Architecture/Helix%203%20Stack.png)
+
+### Preface
+
+Project Helix is based on serverless micro services architecture, where each service is managed in a separate GitHub repository. Each individual service is designed to be single-purpose and can typically be understood on its own within 15 minutes to half an hour. The connections between the different services however that make up the overall architecture require some explanation.
+
+The goal of this document is to provide this explanation and give a lay of the land of the Helix architecture.
+
+The three main areas in Project Helix are:
+
+- **EDGE** delivery through the Fastly Edge computing platform
+- **PIPE** markdown-based content transformation into hypermedia
+- **AUTHOR** authoring of content using Google Docs and Word Online, and publishing into high-availability content stores
+- **DEV** development workflow
+
+The architecture diagram at the beginning of this document uses the Fundamental Modeling Concepts structure chart notation.
+
+Unless otherwise indicated, inter-service communication is based on HTTP.
+
+### EDGE
+
+A typical Helix Pages production deployment uses two stacked Fastly Service Configurations, with each Service Config using a performance-enhancing technique called shielding. 
+
+Shielding means that all traffic from an edge node in Fastly's CDN will be forwarded to a designated "shield" node that then makes requests to the origin or backend server. This increases the likelihood of content already being cached in the shield node and reduces the overall number of requests on the origin. The shield node is simply another edge node that has been picked for reasons of proximity to the origin.
+
+In the architecture chart above, the two CDN configurations are referred to as "outer CDN" (receives incoming requests first, serves a custom domain name, adds custom caching rules) and "inner CDN" (provides basic caching, routes requests to the various backends used by Project Helix)
```

Comment:
`provides basic caching` contradicts `At the Inner CDN level, no caching is applied,` later on.

neither is strictly correct. in the inner CDN we're caching (immutable) media requests only.